parsl/data_provider/files.py

```diff
@@ -28,17 +28,18 @@ class File:
     """
 
     @typeguard.typechecked
-    def __init__(self, url: str):
+    def __init__(self, url: Union[os.PathLike, str]):
         """Construct a File object from a url string.
 
         Args:
-           - url (string) : url string of the file e.g.
+           - url (string or PathLike) : url of the file e.g.
               - 'input.txt'
+              - pathlib.Path('input.txt')
               - 'file:///scratch/proj101/input.txt'
               - 'globus://go#ep1/~/data/input.txt'
               - 'globus://ddb59aef-6d04-11e5-ba46-22000b92c6ec/home/johndoe/data/input.txt'
         """
-        self.url = url
+        self.url = str(url)
         parsed_url = urlparse(self.url)
```

Comment:
There's an __fspath__ dunder-method on PathLike... That can return either bytes or str. And is right next to some functions that encode things to the filesystem encoding, stc
I suspect deeper investigation is an awful can of worms.